include/libtorrent/settings_pack.hpp

```diff
@@ -56,7 +56,7 @@ namespace libtorrent
 	struct settings_pack;
 	struct bdecode_node;
 
-	TORRENT_EXTRA_EXPORT std::shared_ptr<settings_pack> load_pack_from_dict(bdecode_node const& settings);
+	TORRENT_EXTRA_EXPORT void load_pack_from_dict(bdecode_node const& settings, settings_pack& pack);
```

Comment:
I haven't looked through the whole change yet, but if you want to avoid ``shared_ptr``, why not return by value? (i.e. move)